app/forms/state_file/id_disability_form.rb

```diff
@@ -27,10 +27,24 @@ def save
       else
         @intake.update(attributes_for(:intake))
       end
+
+      clean_up_followups
     end
 
     private
 
+    def clean_up_followups
+      primary_followups =  @intake.filer_1099_rs(:primary).map(&:state_specific_followup).compact
+      if primary_disabled == "no" || ["spouse", "none"].include?(mfj_disability)
+        primary_followups.each(&:destroy)
+      end
+
+      if @intake.filing_status_mfj? && ["me", "none"].include?(mfj_disability)
+        spouse_followups =  @intake.filer_1099_rs(:spouse).map(&:state_specific_followup).compact
+        spouse_followups.each(&:destroy)
+      end
+    end
```

Comment:
previously, we kept the followups around if they submitted that they were no longer disabled. The followup only contains what they answer on the next controller (`IdRetirementAndPensionController` and whether `eligible_income_source`). If we do not update the `eligible_income_source` to at least `"unfilled"` or `"no"` the calculator will still consider the 1099R amount as an eligible income (see `calculate_sec_b_line_8e` in ID39R calculator)